solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java

```diff
@@ -3235,6 +3236,14 @@ private static boolean isChildDoc(Object o) {
     return o instanceof SolrInputDocument;
   }
 
+  public static String getUrlFrom(SolrClient solrClient) {
+    if (solrClient instanceof HttpSolrClient) {
```

Comment:
I imagine we would want to check instaceof for other clients too?

Reply:
That was is what I had in mind but then only HttpSolrClient has a baseUrl field including getter and setter methods. Will it be ok to add the same to other clients?

Reply:
lets get some feedback from @dsmiley on our direction!